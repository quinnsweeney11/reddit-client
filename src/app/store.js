import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import postContainerReducer from '../containers/postContainer/postContainerSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    postContainer: postContainerReducer
  },
});
