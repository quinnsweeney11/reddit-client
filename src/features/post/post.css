.post{
    border: 1px solid grey;
    margin-bottom: 15px;
    display:flex;
    justify-content: space-around;
    max-height:400px;
    overflow:scroll;
    background-color:#211d1d;
}

.info {
    margin: 10px 0 10px 10px;
}

.info-item{
    display:inline-block
}

.contents{
    flex-grow:4;
    margin:0 20px;
}

.selftext{
    padding-bottom:20px;
}

a {
    color: #54beff
}

.points {
    color: #FF5700;
}

.children {
    color:#519fcf;
}

.image{
    width:60%;
    padding-bottom:20px;
}

.post_image{
    max-height:none;
    min-height:400px;
}

.image-carousel {
    width: 60%;
}

.carousel{
    width:100%;
}