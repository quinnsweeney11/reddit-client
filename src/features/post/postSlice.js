import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

//Define the inital state of Post:
const initialState = {
    title: '',
    text: '',
    media: [],
    timestamp: '',
    points: '',
    children: ''
}

export const getLocalData = createAsyncThunk('post/fetchLocalData', async () => {
    const response = await fetch('fakedata.json').catch(err => {console.log(err)});
    return response
})


export const postSlice = createSlice({
    name:'post',
    initialState,
    reducers: {
        
    }
})