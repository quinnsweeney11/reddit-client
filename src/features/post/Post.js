import React from "react";
import { useDispatch } from "react-redux";
import "./post.css";
import ArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import ChatBubbleIcon from "@material-ui/icons/ChatBubble";
import { PicCarousel } from "../../utils/picCarousel/PicCarousel";
import "../../../node_modules/video-react/dist/video-react.css";
import ReactHlsPlayer from "react-hls-player";

import { setGallery } from "./postSlice";

//Function that we use to format the text posts.
function htmlDecode(input) {
    var doc = new DOMParser().parseFromString(input, "text/html");

    return doc.documentElement.textContent;
}

export function Post(props) {
    const dispatch = useDispatch();
    if (props.data.secure_media && props.data.secure_media.reddit_video) {
        return (
            <div className="post post_video">
                <div className="info">
                    <div className="info-item points">
                        <ArrowUpwardIcon />
                        <p>{props.data.score}</p>
                    </div>
                    <br />
                    <div className="info-item children">
                        <ChatBubbleIcon />
                        <p>{props.data.num_comments}</p>
                    </div>
                </div>
                <div className="contents">
                    <div className="info-item metadata">
                        <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                        <p><span className="uslash">u/</span>{props.data.author}</p>
                    </div>
                    <h2 className="title-text">{props.data.title}</h2>
                    <ReactHlsPlayer src={props.data.secure_media.reddit_video.hls_url} autoplay={false} controls={true} className="video-container"/>
                </div>
            </div>
        );
    } else if(props.data.secure_media && props.data.secure_media.oembed) {
        return (
            <div className="post post_gif">
                <div className="info">
                    <div className="info-item points">
                        <ArrowUpwardIcon />
                        <p>{props.data.score}</p>
                    </div>
                    <br />
                    <div className="info-item children">
                        <ChatBubbleIcon />
                        <p>{props.data.num_comments}</p>
                    </div>
                </div>
                <div className="contents">
                    <div className="info-item metadata">
                        <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                        <p><span className="uslash">u/</span>{props.data.author}</p>
                    </div>
                    <h2>{props.data.title}</h2>
                    <div className="embedded_media" dangerouslySetInnerHTML={{__html: htmlDecode(props.data.secure_media.oembed.html)}}>
                    </div>
                </div>
            </div>
        )
    }

    if (props.data.post_hint === "image") {
        return (
            <div className="post post_image">
                <div className="info">
                    <div className="info-item points">
                        <ArrowUpwardIcon />
                        <p>{props.data.score}</p>
                    </div>
                    <br />
                    <div className="info-item children">
                        <ChatBubbleIcon />
                        <p>{props.data.num_comments}</p>
                    </div>
                </div>
                <div className="contents">
                    <div className="info-item metadata">
                        <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                        <p><span className="uslash">u/</span>{props.data.author}</p>
                    </div>
                    <h2>{props.data.title}</h2>
                    <img
                        className="image"
                        alt="Post Content"
                        src={props.data.url}
                    />
                </div>
            </div>
        );
    }

    if (props.data.gallery_data) {
        return (
            <div className="post post_carousel">
                <div className="info">
                    <div className="info-item points">
                        <ArrowUpwardIcon />
                        <p>{props.data.score}</p>
                    </div>
                    <br />
                    <div className="info-item children">
                        <ChatBubbleIcon />
                        <p>{props.data.num_comments}</p>
                    </div>
                </div>
                <div className="contents">
                    <div className="info-item metadata">
                        <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                        <p><span className="uslash">u/</span>{props.data.author}</p>
                    </div>
                    <h2>{props.data.title}</h2>
                    <PicCarousel
                        className="carousel"
                        pics={props.data.gallery_data.items}
                    />
                </div>
            </div>
        );
    }

    if (props.data.selftext === "") {
        return (
            <div className="post">
                <div className="info">
                    <div className="info-item points">
                        <ArrowUpwardIcon />
                        <p>{props.data.score}</p>
                    </div>
                    <br />
                    <div className="info-item children">
                        <ChatBubbleIcon />
                        <p>{props.data.num_comments}</p>
                    </div>
                </div>
                <div className="contents">
                    <div className="info-item metadata">
                        <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                        <p><span className="uslash">u/</span>{props.data.author}</p>
                    </div>
                    <h2>{props.data.title}</h2>
                </div>
            </div>
        );
    }

    return (
        <div className="post">
            <div className="info">
                <div className="info-item points">
                    <ArrowUpwardIcon />
                    <p>{props.data.score}</p>
                </div>
                <br />
                <div className="info-item children">
                    <ChatBubbleIcon />
                    <p>{props.data.num_comments}</p>
                </div>
            </div>
            <div className="contents">
                <div className="info-item metadata">
                    <p><span className="rslash">r/</span>{props.data.subreddit}</p>
                    <p><span className="uslash">u/</span>{props.data.author}</p>
                </div>
                <h4>{props.data.title}</h4>
                <div
                    className="selftext"
                    dangerouslySetInnerHTML={{
                        __html: htmlDecode(props.data.selftext_html),
                    }}
                ></div>
            </div>
        </div>
    );
}
