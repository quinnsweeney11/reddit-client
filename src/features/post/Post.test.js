import React from "react";
import { shallow } from "enzyme";
import { Post } from "./Post";
import renderer from "react-test-renderer";
import { imgPost, txtPost, vidPost } from "../../fakeData/post";
import { Provider } from "react-redux";
import { store } from "../../app/store";

describe("renders", () => {
  it("photo post", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Post data={imgPost.data} />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("text post", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Post data={txtPost.data} />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("video post", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Post data={vidPost.data} />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
