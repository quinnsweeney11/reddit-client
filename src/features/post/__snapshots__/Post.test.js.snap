// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot photo post 1`] = `
<div
  className="post post_image"
>
  <div
    className="info"
  >
    <div
      className="info-item points"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
        />
      </svg>
      <p>
        35744
      </p>
    </div>
    <br />
    <div
      className="info-item children"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2z"
        />
      </svg>
      <p>
        407
      </p>
    </div>
  </div>
  <div
    className="contents"
  >
    <div
      className="info-item metadata"
    >
      <p>
        <span
          className="rslash"
        >
          r/
        </span>
        funny
      </p>
      <p>
        <span
          className="uslash"
        >
          u/
        </span>
        LordRupertEverton84
      </p>
    </div>
    <h2>
      My daughter won our town's "Name the Street Sweeper" contest.
    </h2>
    <img
      alt="Post Content"
      className="image"
      src="https://i.redd.it/isbqfdsqgcy61.jpg"
    />
  </div>
</div>
`;

exports[`snapshot text post 1`] = `
<div
  className="post"
>
  <div
    className="info"
  >
    <div
      className="info-item points"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
        />
      </svg>
      <p>
        12796
      </p>
    </div>
    <br />
    <div
      className="info-item children"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2z"
        />
      </svg>
      <p>
        2088
      </p>
    </div>
  </div>
  <div
    className="contents"
  >
    <div
      className="info-item metadata"
    >
      <p>
        <span
          className="rslash"
        >
          r/
        </span>
        Superstonk
      </p>
      <p>
        <span
          className="uslash"
        >
          u/
        </span>
        timmmmmmmyy
      </p>
    </div>
    <h4>
      "Glacier Capital" does not exist.
    </h4>
    <div
      className="selftext"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<!-- SC_OFF --><div class=\\"md\\"><p>My friends, in all the mocking of Glacier Capital shorting GME today, we forgot to ask the most important question: do they actually exist? Bloomberg has a <a href=\\"https://www.bloomberg.com/profile/company/566411Z:US\\">profile</a> of them, if you can call it that, claiming they were founded 3/14/05, but outside of this I can find no evidence that they existed until recently. No article mentions, no chest puffing PR, nothing but their linkedIn pages and their website.</p>

<p>&quot;Glacier Capital&quot; on their <a href=\\"http://glaciercapital.lu/\\">potentially insecure website</a> lists an address of record of <a href=\\"https://imgur.com/XyuT7S3\\">44 Boulevard de Verdun, L-2670 Luxembourg.</a> Now I don&#39;t know about you, but to me, that looks like an apartment building, not a multimillion dollar hedge fund. </p>

<p>So, I called &quot;Glacier.&quot; As we have previously observed, Melvin Capital doesn&#39;t even answer the phone anymore. Citadel at least still has the automated system pick up the phone. But in the case of &quot;Glacier Capital&quot; there is no automated system. Instead, we appear to be calling a personal cell phone, with the voice mailbox of &quot;Alex Larsen.&quot; Don&#39;t believe me? Call yourself. &quot;Glacier Capital&#39;s&quot; public phone number is +352 661 932 584. If you&#39;re in the US or Canada, you&#39;ll have to dial a 011 first to get into the international switching system.</p>

<p>I challenge anyone to prove Glacier Capital exists.</p>

<p>TL;DR &quot;Glacier Capital&quot; is FUD, they don&#39;t exist except on paper.</p>

<p>Edit: One more point of evidence in favor of this being a personal cell phone: If you call once, it goes straight to voicemail, but if you immediately call back it rings several times before going to voicemail. Android has a setting to allow calls to punch through &quot;Do not disturb&quot; mode if the same person calls twice in 5 minutes, which may be exactly what&#39;s happening here.</p>

<p>Edit PS: Credit to <a href=\\"https://old.reddit.com/r/Superstonk/comments/new7b6/check_out_glacier_capitals_head_office_no/\\">this post</a> for inspiring this investigation.</p>

<p>Edit 2: As <a href=\\"/u/mrperson221\\">u/mrperson221</a> said, why would a fund publicly advertise opening a new short position, especially in a stock that has been so volatile and known for retailers jumping on the bandwagon to crush short sellers? It paints a giant target on them. &quot;Glacier Capital&quot; is 100% FUD.</p>

<p>Edit 3: As some have commented, the owner of that cell phone seems to have turned it off or changed the settings, as the second call in a row is no longer punching through and ringing.</p>

<p>Edit 4: It appears the google image results may just be a google error since the searched picture doesn&#39;t seem to appear on the searched pages. I have no idea why they&#39;re showing up as matches or why similar results are not shown when searching the pictures of the other members of their staff.</p>

<p>Edit 5: To be clear, I do not mean to imply that there is no such legal entity as Glacier Capital. There are <a href=\\"https://www.lbr.lu/mjrcs/jsp/DisplayConsultDetailCompanyActionNotSecured.action?time=1621313108273&amp;CURRENT_TIMESTAMP_ID=1621313090005#null\\">public records</a> that show that such an entity does exist. What I am doubting is whether they are actually what they claim to be, an independent investment company, or whether they are a facade for some other entity.</p>

<p>Edit 6: Glacier Capital&#39;s website is not using https, so maybe save yourself the click unless you have to and clear your cookies afterward.</p>
</div><!-- SC_ON -->",
        }
      }
    />
  </div>
</div>
`;

exports[`snapshot video post 1`] = `
<div
  className="post post_video"
>
  <div
    className="info"
  >
    <div
      className="info-item points"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
        />
      </svg>
      <p>
        61681
      </p>
    </div>
    <br />
    <div
      className="info-item children"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2z"
        />
      </svg>
      <p>
        1179
      </p>
    </div>
  </div>
  <div
    className="contents"
  >
    <div
      className="info-item metadata"
    >
      <p>
        <span
          className="rslash"
        >
          r/
        </span>
        aww
      </p>
      <p>
        <span
          className="uslash"
        >
          u/
        </span>
        jackennethfn
      </p>
    </div>
    <h2
      className="title-text"
    >
      Thatâ€™s probably the best day that mouse has had or will have in its life.
    </h2>
    <video
      autoplay={false}
      className="video-container"
      controls={true}
      src="https://v.redd.it/wlle3q7kwrz61/HLSPlaylist.m3u8?a=1623909969%2CNjQ4MzYyNDc3MTU1MGY5OWZkNGY4NjBkMDdjNTYyZTBhMmUzNjkxYTA5MzEwODRjMGQxYTU3ZTI5OWU3Y2VlYw%3D%3D&amp;v=1&amp;f=sd"
    />
  </div>
</div>
`;
